src/System.Management.Automation/engine/Utils.cs

```diff
@@ -478,13 +478,22 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
         }
 #endif
 
-        internal static string DefaultPowerShellAppBase => GetApplicationBase(DefaultPowerShellShellID);
+        internal static string DefaultPowerShellAppBase => GetApplicationBase();
 
-        internal static string GetApplicationBase(string shellId)
+        internal static string GetApplicationBase()
         {
-            // Use the location of SMA.dll as the application base.
-            Assembly assembly = typeof(PSObject).Assembly;
-            return Path.GetDirectoryName(assembly.Location);
+            // AppContext is needed where PS7 may be compiled as a single exe
+            // Environment.ProcessPath covers the general case
```

Comment:
What is the general case that `Environment.ProcessPath` is going to cover?
`AppContext.BaseDirectory` calls `Environment.ProcessPath` in case of NativeAOT, so I think that already covers it, isn't it?